osu.Game.Rulesets.Osu/OsuInputManager.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Input;
+using osu.Game.Input;
+using OpenTK.Input;
+using KeyboardState = osu.Framework.Input.KeyboardState;
+using MouseState = osu.Framework.Input.MouseState;
+
+namespace osu.Game.Rulesets.Osu
+{
+    public class OsuInputManager : ActionMappingInputManager<OsuAction>
+    {
+        public OsuInputManager(RulesetInfo ruleset) : base(ruleset, concurrencyMode: ConcurrentActionMode.UniqueActions)
+        {
+
+        }
+        protected override void TransformState(InputState state)
+        {
+            base.TransformState(state);
+
+            var mouse = state.Mouse as MouseState;
+            var keyboard = state.Keyboard as KeyboardState;
+
+            if (mouse != null && keyboard != null)
+            {
+                if (mouse.IsPressed(MouseButton.Left))
+                    keyboard.Keys = keyboard.Keys.Concat(new[] { Key.LastKey + 1 });

```

Comment:
this is obviously temporary. we probably want to make our own framework-level `Key` enum or consider other options for extensibility.